particle-iot/device-os
# Release timer 2, use systick to calculate debounce time

### Problem

Button driver occupied a high precision to calculate debounce time,  it is wasteful, we can use systick handler to calculate debounce time.

### Steps to Test
1. Flash example application
2. Press the mode button, observe the press count and debounce time.

### Example App

```c
 #include "application.h"
 #include "button_hal.h"
 
SYSTEM_MODE(MANUAL);

Serial1LogHandler logHandler(115200);

static volatile int counter = 0;

void setup() {
  System.on(button_click, [](system_event_t ev, int clicks) -> void {
    counter++;
  });
}

void loop() {
    Log.info("counter: %d, debounced time: %d", counter, BUTTON_GetDebouncedTime(BUTTON1));
    delay(1000);
}
```

### References



---

### Completeness
